Adding information about using GitHub and GitHub Desktop

vickytnz opened the pull request:
Ported over the GOV.UK Prototype Kit guides:
1. Store your code online with GitHub and GitHub Desktop
2. Store your new prototype in GitHub
3. Download an existing prototype from GitHub4. Collaborate on prototypes using GitHub Desktop

Reformatted the guides page to have a section.

I do wonder if it would be better to reconfigure the install section to just be about Github or somehow pull the downloading someone else's prototype earlier with an easier through-line.  

This could also align with @paulmsmith's work on Codespaces.

Pages:

How to (changed to 3 columns) and with some new ideas
 
![Guides page with section 'Version control and collaboration' with pages 'Store your code online with GitHub and GitHub Desktop', 'Store your new prototype in GitHub', 'Download an existing prototype from GitHub', 'Collaborate on prototypes using GitHub Desktop'](https://github.com/user-attachments/assets/aa515b75-1b3b-4c13-afcc-58ca53f5affa)

## Store your code online with GitHub and GitHub Desktop

![Store your code online with GitHub and GitHub Desktop page](https://github.com/user-attachments/assets/89530183-c149-422d-b092-1ed02aeda853)

##  Store your new prototype in GitHub

![Store your new prototype in GitHub page](https://github.com/user-attachments/assets/34ba4034-9607-412d-adda-2811868861b0)

## Download an existing prototype from GitHub
![Download an existing prototype from GitHub](https://github.com/user-attachments/assets/33aa681b-bb13-4fbb-b5e4-24bfe7e9f27a)

## Collaborate on prototypes using GitHub Desktop

![Collaborate on prototypes using GitHub Desktop page](https://github.com/user-attachments/assets/f1af82e6-f2af-4691-897e-73184fa799f2)

vickytnz commented:
I'm also now wondering how much this needs to be "NHSed" to at least note that this should usually be done with official NHS repositories and with our processes. DfE does similar with their guidance for design ops

https://design.education.gov.uk/design-system/github-heroku

vickytnz commented:
Version now ready for review

## How to page with new section
![How to with Storing and sharing code, Store your code online with GitHub and GitHub Desktop, Using GitHub if you work for NHS England, Store your new prototype in GitHub, Download an existing prototype from GitHub, Collaborate on prototypes using GitHub Desktop, Setting up Git using the terminal](https://github.com/user-attachments/assets/b14b39e2-6a6c-4036-b5d4-19558ed33c21)

##  Store your code online with GitHub and GitHub Desktop
Similar to GOV.UK page but some tweaking and addition of NHS England info and a course correct to start with installing as needed
![Store your code online with GitHub and GitHub Desktop](https://github.com/user-attachments/assets/c865a622-9862-4949-bc4a-97f0a829c068)


## Using GitHub if you work for NHS England
NEW PAGEThis is trying to have just enough information without duplicating stuff that is in Sharepoint organisations. Main thing is to encourage people to set up Github early rather than go rogue. 
![Using GitHub if you work for NHS England](https://github.com/user-attachments/assets/dd733853-7227-492a-8725-89ea0d32b5d3)


## Store your new prototype in GitHub
As per the GOV.UK version with extra linking to other sections.
![Store your new prototype in GitHub](https://github.com/user-attachments/assets/fa83dfd1-b3f3-45b4-a030-e56b075ad693)

## Download an existing prototype from GitHub
As per the GOV.UK version with extra linking to other sections.
![Download an existing prototype from GitHub](https://github.com/user-attachments/assets/7bf4970c-1651-4779-87cd-f1988cd7422f)


## Collaborate on prototypes using GitHub Desktop
As per the GOV.UK version. This hasn't been changed. 
![Collaborate on prototypes using GitHub Desktop](https://github.com/user-attachments/assets/26f46b81-f067-4267-a068-6b07d4238670)

edwardhorsford commented on app/views/how-tos/github/collaborate.html:
```diff
@@ -0,0 +1,130 @@
+{% extends "layout.html" %} {% block pageTitle %} Collaborate on prototypes
+using GitHub Desktop - NHS prototype kit {% endblock %} {% block beforeContent
+%} {{ breadcrumb({ items: [ { href: "/", text: "Home" }, { href: "/how-tos",
+text: "Guides" }, { href: "/how-tos/github/index", text: "Store code online with
+GitHub and GitHub desktop" } ] }) }} {% endblock %} {% block content %}
```
Could this get spread out? it's quite hard to read.

vickytnz replied:
gah this was prettifier not dealing well with nunjucks and me being lazy. fixed

edwardhorsford commented on app/views/how-tos/github/download-existing-prototype.html:
```diff
@@ -0,0 +1,61 @@
+
+    <p>
+      To store your new prototype in GitHub, you need to
+      <a href="index">set up GitHub Desktop</a>. If you work for NHS England,
```
Doubt we want to link to a url with `index`?

vickytnz replied:
changed to absolute url to start of section

edwardhorsford commented on app/views/how-tos/github/index.html:
```diff
@@ -0,0 +1,95 @@
+      </li>
+    </ul>
+
+    <div class="nhsuk-inset-text">
+      <span class="nhsuk-u-visually-hidden">Information: </span>
+      <p>
+        If you have never run the NHS prototype kit on your computer, finish the
+        <a href="/install/index">get started guide</a> first. This will mean
+        that you have the right tools to complete these steps.
+      </p>
+    </div>
```
Use the inset text component?

vickytnz replied:
Changed

edwardhorsford commented on app/views/how-tos/github/nhs-england.html:
```diff
@@ -0,0 +1,49 @@
+      <li>make changes to an existing prototype</li>
+    </ul>
+
+    <div class="nhsuk-inset-text">
+      <span class="nhsuk-u-visually-hidden">Information: </span>
+      <p>
+        Some NHS prototype repositories are in different GitHub organisations.
+        These repositories are being merged - do not add new repositories to
+        them unless your team tells you to do it.
+      </p>
+    </div>
```
Use component

edwardhorsford commented on app/views/how-tos/github/store-new-prototype.html:
```diff
@@ -0,0 +1,81 @@
+
+    <p>
+      To store your new prototype in GitHub, you need to
+      <a href="index">set up GitHub Desktop</a>. If you work for NHS England,
```
Probably don't want to link to index

vickytnz replied:
I was being lazy and using relative urls - have changed to absolute (it's the index page for this section, not the home page)

edwardhorsford commented on app/views/how-tos/github/store-new-prototype.html:
```diff
@@ -0,0 +1,81 @@
+    <ol class="nhsuk-list nhsuk-list--number">
+      <li>In the top right, click <strong>publish repository</strong>.</li>
+      <li>
+        Make sure '<strong>Keep this code private'</strong> is not ticked, as we
```
Inconsistent positioning of strong tag

vickytnz replied:
Good catch - fixed

edwardhorsford commented on app/views/how-tos/index.html:
```diff
@@ -91,15 +91,37 @@ <h2 class="nhsuk-heading-s">General use</h2>
 
       <ul class="nhsuk-list">
-        <li><a href="/how-tos/git">Setting up Git</a></li>
+        <li><a href="/how-tos/github/index">Store your code online with GitHub and GitHub Desktop</a>
```
probably wants to link to `how-tos/github`

vickytnz replied:
changed (along with ordering as this is going to the old guidance for just git as a concept) - i'm leaving that page in for now to decide whether it's worth taking out or not

vickytnz commented:
Latest version included comment fixes but also uses the 'mini-hub component. This makes it easier to see the steps (though it does make me wonder if the NHS England one is in the right place) 

![page with mini-hub layout](https://github.com/user-attachments/assets/fdf257a8-e4c3-46f7-b532-2516fecbcea7)

I have also noticed that this pattern is different from the one for the basic prototype which is more like a transactional thing (namely the title hint has switched from above to below) - not sure if this needs to get fixed

vickytnz commented:
After more reviews I've done a complete redo to:

Redo the how-to page

<img width="322" alt="Storing and sharing code
Store your code online with GitHub and GitHub Desktop
Set up GitHub and GitHub Desktop
Store a new prototype in GitHub
Download an existing prototype from GitHub
Collaborate on prototypes using GitHub Desktop
Set up Git using the terminal" src="https://github.com/user-attachments/assets/765f92d8-b9ad-4f3d-a2e1-070e0d870860">

## Overview page (new)
This is now just about the basic stuff but also links over to the git terminal information
![page with concepts and redirect to git on terminal](https://github.com/user-attachments/assets/f11e76f7-3d52-47f9-bb0c-8e081aa753dc)

## Set up GitHub and GitHub Desktop
This page:
* takes what was the second part of the initial setup page 
* added details about installing git as that was missing (PS I hope these instructions work)
* merges the NHS England stuff and cuts it down a lot (I've removed stuff about other orgs and other repos) 

![set up page with install git, get github account, and download and setup github desktop](https://github.com/user-attachments/assets/4c0bddae-e3ee-48fd-88b4-323f3df26497)


##  Store a new prototype on GitHub
This page adds course corrects to other starting points (new prototype and the github setup)

![store a new prototype on GitHub page](https://github.com/user-attachments/assets/c25f50fd-e35b-4c86-85d8-12b226ce0f5e)

## Download an existing prototype from GitHub 

Linking to previous pages 

![Download an existing prototype from GitHub page](https://github.com/user-attachments/assets/0166e2bf-acbc-4849-a3f0-dc01c09b89e3)

## Collaborate on a prototype
![Collaborate on a prototype](https://github.com/user-attachments/assets/ba891586-d108-4d36-a3cd-ba95f6c8263c)

Only remaining things for me are:
- consistency of titles (maybe it's overkill on the Github stuff and titles could be just 'Download an existing prototype' etc)
- overall consistency of titles with other sections (e.g. the subheader is different here to the basic prototype)

frankieroberto commented:
@sarawilcox would you mind reviewing this too?

sarawilcox commented:
Hi @frankieroberto , I've reviewed this and have made a few tiny tweaks. Probably best to publish it as is. I haven't had a chance to test the process myself.

The main change I made was to make people aware that there are 2 main GitHub orgs for NHSE: the NHS Digital one and the NHS.UK one. (There may be more for all we know.)

I think we need to review this when we look at publishing content about CodeSpace - both where it sits in the IA and how we document some content (e.g. commands) that will be common to both.

sarawilcox commented:
I did notice that <code>main</code>, for example, displays differently from how it does in the design system. This way of displaying it wasn't very obvious to me. 

![Screenshot 2025-02-10 at 13 13 14](https://github.com/user-attachments/assets/9f635d00-3cf6-43d3-9d95-dddd98f4cfcb)

frankieroberto commented:
> I did notice that `<code>main</code>`, for example, displays differently from how it does in the design system. This way of displaying it wasn't very obvious to me.

Good point. I think we should tackle this as part of #21 and make the styles for code examples consistent with the service manual. 👍
